src/java.base/share/classes/java/util/Currency.java

```diff
@@ -462,53 +464,52 @@ public static Set<Currency> getAvailableCurrencies() {
      * @since 25
      */
     public static Stream<Currency> availableCurrencies() {
-        return getCurrencies().stream();
+        return available.get().stream();
     }
 
-    // Returns the set of available Currencies which are lazily initialized
-    private static synchronized HashSet<Currency> getCurrencies() {
-        if (available == null) {
-            var sysTime = System.currentTimeMillis();
-            available = HashSet.newHashSet(256);
-
-            // Add simple currencies first
-            for (char c1 = 'A'; c1 <= 'Z'; c1 ++) {
-                for (char c2 = 'A'; c2 <= 'Z'; c2 ++) {
-                    int tableEntry = getMainTableEntry(c1, c2);
-                    if ((tableEntry & COUNTRY_TYPE_MASK) == SIMPLE_CASE_COUNTRY_MASK
-                            && tableEntry != INVALID_COUNTRY_ENTRY) {
-                        char finalChar = (char) ((tableEntry & SIMPLE_CASE_COUNTRY_FINAL_CHAR_MASK) + 'A');
-                        int defaultFractionDigits = (tableEntry & SIMPLE_CASE_COUNTRY_DEFAULT_DIGITS_MASK) >> SIMPLE_CASE_COUNTRY_DEFAULT_DIGITS_SHIFT;
-                        int numericCode = (tableEntry & NUMERIC_CODE_MASK) >> NUMERIC_CODE_SHIFT;
-                        StringBuilder sb = new StringBuilder();
-                        sb.append(c1);
-                        sb.append(c2);
-                        sb.append(finalChar);
-                        available.add(getInstance(sb.toString(), defaultFractionDigits, numericCode));
-                    } else if ((tableEntry & COUNTRY_TYPE_MASK) == SPECIAL_CASE_COUNTRY_MASK
-                            && tableEntry != INVALID_COUNTRY_ENTRY
-                            && tableEntry != COUNTRY_WITHOUT_CURRENCY_ENTRY) {
-                        int index = SpecialCaseEntry.toIndex(tableEntry);
-                        SpecialCaseEntry scEntry = specialCasesList.get(index);
-
-                        if (scEntry.cutOverTime == Long.MAX_VALUE
-                                || sysTime < scEntry.cutOverTime) {
-                            available.add(getInstance(scEntry.oldCurrency,
-                                    scEntry.oldCurrencyFraction,
-                                    scEntry.oldCurrencyNumericCode));
-                        } else {
-                            available.add(getInstance(scEntry.newCurrency,
-                                    scEntry.newCurrencyFraction,
-                                    scEntry.newCurrencyNumericCode));
-                        }
+    // Builds and returns the set of available Currencies
+    private static HashSet<Currency> getCurrencies() {
```

Comment:
I think we should rename this to `computeCurrencies` or `makeCurrencies` to indicate we are creating a new instance.

Reply:
Definitely; updated in https://github.com/openjdk/jdk/pull/25134/commits/e2672ba1846c17b723d6bcbee0ef20488c0dca24.